diag_manager/fms_diag_field_object.F90

```diff
@@ -406,6 +408,24 @@ subroutine set_vartype(objin , var)
  end select
 end subroutine set_vartype
 
+!> @brief Sets the send data time
+subroutine set_send_data_time (this, time)
+  class (fmsDiagField_type) , intent(inout):: this                !< The field object
+  type(time_type),            intent(in)   :: time                !< Current model time
+
+  call this%input_data_buffer%set_send_data_time(time)
+end subroutine set_send_data_time
+
+!> @brief Get the send data time
+!! @result Send data time
```

Comment:
Could you make the description of what "send data time" is a little more detailed. Something like "The time that was send by the model component to send_data".  You have a better description in `fmsDiagInputBuffer_t`